examples/quantum_annealing/price_optimization/qubo_dynamic_pricing.py

```diff
@@ -0,0 +1,256 @@
+    d = b
+    for i in range(len(coeff)):
+        d += coeff[i]*prices[i]
+
+    return d
+
+
+def get_variance(data_x, p, sigma):
+    """
+    :param data_x (np.array): [n_samples, n_days]
+    :param p (list): [n_days]
+    :return: variance
+    """
+    n_samples, t = data_x.shape
+    ones = np.ones((n_samples, 1), dtype=np.float)
+    x_mat = np.concatenate([ones, data_x], axis=1)  # [n_samples, n_days+1]
+    x_mat = np.linalg.inv(
+        np.dot(x_mat.T, x_mat)
+    )
+    p = np.array([1.]+p)
+    variance = (sigma**2) * (1. + p.dot(x_mat).dot(p))
+    return variance
+
+
+def create_program(a,b, p_data, price_levels, data_x, Lp, Ld, sigma, beta, vol_bound):
+    """
+
+    :param a (list of int): [7], coefficient
+    :param b (int):
+    :param p_data (list of int): [7], past 7 days prices
+    :param price_levels (list of int): number of price levels
+    :param data_x (np.array): [nsamples, n_days]
+    :param L (float): coeff of constraints penalty
+    :param sigma (float): standard deviation of noise
+    :param beta (float): coeff of variance
+    :return:
+    """
+    assert type(a) is list
+    #assert type(b) is int
+    assert type(p_data) is list
+    assert type(price_levels) is list
+
+    t = len(a)
+    n_level = len(price_levels)
+
+    # variables
+    x = []
+    p = []
+    d = []
+
+    # get p
+    for i in range(t):
+        p_i = 0
+        for j in range(n_level):
+            x_ij = Binary(f"X_{i*n_level+j:03d}")
+            x.append(x_ij)
+            p_i += x_ij*price_levels[j]
+        p.append(p_i)
+
+    all_p = p_data + p
+
+    # get d, rev
+    rev = 0
+    for i in range(t):
+        d_i = get_demand(
+            coeff=a,
+            b=b,
+            prices=all_p[i+1:i+1+t]
+        )
+        d.append(d_i)
+        rev += d_i * p[i]
+        # minus variance
+        rev -= beta * get_variance(data_x, all_p[i+1:i+1+t], sigma)
+        # add inequaliry constraints
+        if vol_bound:
+            _, d_const = get_coeffient(d_i, 'linear')
+            rev -= inequality_penalty(
+                demand=d_i,
+                demand_name=f'demand{i}',
+                vol_bounday=vol_bound,
+                d_const=d_const,
+                Ld=Ld
+            )
+
+    # add equalty constraints
+    for i in range(t):
+        penalty = x[i*n_level]
+        for j in range(1, n_level):
+            penalty += x[i*n_level+j]
+        penalty = ((penalty-1)**2)*Lp
+        rev -= penalty
+
+    return rev, d
+
+
+def construct_slack(n, name):
+    e = 0
+    slack = 0
+    while n >= 2**e:
+        n -= 2**e
+        slack += (2**e) * Binary(f"{name}_{e}")
+        e += 1
+
+    slack += n * Binary(f"{name}_{e}")
+    return slack
+
+
+def inequality_penalty(demand, demand_name, vol_bounday, d_const, Ld):
+    rhs = vol_bounday - d_const
+    assert rhs <= 0
+    n = -rhs
+    slack = construct_slack(n, demand_name)
+    return ((demand-vol_bounday-slack)**2)*Ld
+
+
+def main(
```

Comment:
nit: Should we have a different function name instead of `main`?

Reply:
yes, now we change the name of `main` to `optimize`